python/instrumentation/openinference-instrumentation-langchain/tests/openinference/instrumentation/langchain/test_instrumentor.py

```diff
@@ -0,0 +1,353 @@
+    assert llm_attributes.pop(OPENINFERENCE_SPAN_KIND, None) == CHAIN.value
+    assert llm_attributes.pop(INPUT_VALUE, None) is not None
+    assert llm_attributes.pop(INPUT_MIME_TYPE, None) == JSON.value
+    if status_code == 200:
+        assert llm_attributes.pop(OUTPUT_VALUE, None) == output_messages[0]["content"]
+    elif status_code == 400:
+        assert llm_span.status.status_code == trace_api.StatusCode.ERROR
+        assert llm_span.events[0].name == "exception"
+        assert (llm_span.events[0].attributes or {}).get(
+            OTELSpanAttributes.EXCEPTION_TYPE
+        ) == "BadRequestError"
+    assert llm_attributes == {}
+
+    assert (oai_span := spans_by_name.pop("ChatOpenAI", None)) is not None
+    assert oai_span.parent is not None
+    assert oai_span.parent.span_id == llm_span.context.span_id
+    assert oai_span.context.trace_id == llm_span.context.trace_id
+    oai_attributes = dict(oai_span.attributes or {})
+    assert oai_attributes.pop(OPENINFERENCE_SPAN_KIND, None) == LLM.value
+    assert oai_attributes.pop(LLM_MODEL_NAME, None) is not None
+    assert oai_attributes.pop(LLM_INVOCATION_PARAMETERS, None) is not None
+    assert oai_attributes.pop(INPUT_VALUE, None) is not None
+    assert oai_attributes.pop(INPUT_MIME_TYPE, None) == JSON.value
+    assert oai_attributes.pop(LLM_PROMPTS, None) is not None
+    if status_code == 200:
+        assert oai_span.status.status_code == trace_api.StatusCode.OK
+        assert oai_attributes.pop(OUTPUT_VALUE, None) is not None
+        assert oai_attributes.pop(OUTPUT_MIME_TYPE, None) == JSON.value
+        assert (
+            oai_attributes.pop(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_ROLE}", None)
+            == output_messages[0]["role"]
+        )
+        assert (
+            oai_attributes.pop(f"{LLM_OUTPUT_MESSAGES}.0.{MESSAGE_CONTENT}", None)
+            == output_messages[0]["content"]
+        )
+        if not is_stream:
+            assert (
+                oai_attributes.pop(LLM_TOKEN_COUNT_TOTAL, None) == completion_usage["total_tokens"]
+            )
+            assert (
+                oai_attributes.pop(LLM_TOKEN_COUNT_PROMPT, None)
+                == completion_usage["prompt_tokens"]
+            )
+            assert (
+                oai_attributes.pop(LLM_TOKEN_COUNT_COMPLETION, None)
+                == completion_usage["completion_tokens"]
+            )
+    elif status_code == 400:
+        assert oai_span.status.status_code == trace_api.StatusCode.ERROR
+        assert oai_span.events[0].name == "exception"
+        assert (oai_span.events[0].attributes or {}).get(
+            OTELSpanAttributes.EXCEPTION_TYPE
+        ) == "BadRequestError"
+    assert oai_attributes == {}
+
+    # The remaining span is from the openai instrumentor.
+    openai_span = spans_by_name.popitem()[1]
+    assert openai_span.parent is not None
+    if is_async:
+        # FIXME: it's unclear why the context fails to propagate.
```

Comment:
Maybe this is a TODO and a follow-up ticket rather than a FIXME

Reply:
seems like a FIXME because it's not inline with expectation, where TODO is more like extra features or something